x/cosmwasmpool/model/codec.go

```diff
@@ -0,0 +1,54 @@
+package model
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+
+	"github.com/osmosis-labs/osmosis/v15/x/cosmwasmpool/types"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v15/x/poolmanager/types"
+)
+
+func RegisterCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&CosmWasmPool{}, "osmosis/cw-pool", nil)
+	cdc.RegisterConcrete(&MsgCreateCosmWasmPool{}, "osmosis/cw-create-pool", nil)
+}
+
+func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"osmosis.poolmanager.v1beta1.PoolI",
+		(*poolmanagertypes.PoolI)(nil),
+		&CosmWasmPool{},
+	)
+	registry.RegisterInterface(
+		"osmosis.cosmwasmpool.v1beta1.CosmWasmExtension",
+		(*types.CosmWasmExtension)(nil),
+		&CosmWasmPool{},
+	)
+
+	registry.RegisterImplementations(
+		(*sdk.Msg)(nil),
+		&MsgCreateCosmWasmPool{},
+	)
+
+	msgservice.RegisterMsgServiceDesc(registry, &_MsgCreator_serviceDesc)
+}
+
+var (
+	amino     = codec.NewLegacyAmino()
+	ModuleCdc = codec.NewAminoCodec(amino)
+)
+
+func init() {
+	// TODO: uncomment the lines below: https://github.com/osmosis-labs/osmosis/issues/5329
```

Comment:
I don't see this TODO in the tracking issue

Reply:
I believe it's the third one: "Uncomment lines in codec.go for cw pool model"